scripts/fgi/run

```diff
@@ -244,6 +245,7 @@ if args.report:
             if read_lines:
                 test_report.append(line)
             if 'json-report-begin' in line: read_lines = True
+            if 'test result: ok' in line: test_res = 'passed'
```

Comment:
Zihao and I discussed this offline. Sometimes the cluster cleanup process that occurs on "Drop" doesn't complete properly (e.g. failed healthcheck). This results in a nonzero exit code despite the Forge tests themselves running fine. The cluster then repairs itself on the next run

I think a cleaner way to go about this would be to have `fgi` parse a JSON report generated by the `forge`, like how Cluster Test did. We could then record more specific metrics from test output and cluster spin up/down operations.

Reply:
yea, from fgi script we do not have ability to grant exit code for some cases. Add more fileds in JSON report is the next step I am going to do, which allows us have different/separate to determine actual tests failures or infra failures

Reply:
Perfect, thanks for the explanation!